actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -32,29 +32,22 @@ class InvalidCrossOriginRequest < ActionControllerError #:nodoc:
   # response may be extracted. To prevent this, only XmlHttpRequest (known as XHR or
   # Ajax) requests are allowed to make requests for JavaScript responses.
   #
-  # It's important to remember that XML or JSON requests are also checked by default. If
-  # you're building an API or an SPA you could change forgery protection method in
-  # <tt>ApplicationController</tt> (by default: <tt>:exception</tt>):
+  # Subclasses of <tt>ActionController::Base</tt> are protected by default with the
+  # <tt>:exception</tt> strategy, which raises an
+  # <tt>ActionController::InvalidAuthenticityToken</tt> error on unverified requests.
+  #
+  # It's important to remember that XML or JSON requests are also checked by default.
+  # APIs may want to disable this behavior since they are typically designed to be
+  # state-less: that is, the request API client handles the session instead of Rails.
+  # One way to achieve this is to use the <tt>:null_session</tt> strategy instead,
+  # which allows unverified requests to be handled, but with an empty session:
   #
   #   class ApplicationController < ActionController::Base
-  #     protect_from_forgery unless: -> { request.format.json? }
+  #     protect_from_forgery with: :null_session
   #   end
   #
-  # It is generally safe to exclude XHR requests from CSRF protection
-  # (like the code snippet above does), because XHR requests can only be made from
-  # the same origin. Note however that any cross-origin third party domain
```

Comment:
Think we should keep around this warning on cors.

Reply:
It's only relevant if you disable CSRF protection, which I was trying to avoid mentioning at all... it also only makes sense for requests which would otherwise be unaffected by CSRF.

That's the case when `xhr?` returns true, since the `X-Requested-With` header will force a CORS preflight. It's _not_ the case for XHR requests is general, and I worry that people will confuse the two. I also think it's too easy to add CORS headers later, months or years after you read this documentation, and introduce a vulnerability.